sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DontMixIncrementOrDecrementWithOtherOperators.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Program
+    {
+        public void Foo()
+        {
+            int val1 = 0;
+            int val2 = 0;
+
+            var result = ++val1 + val2; // Noncompliant {{Extract this [increment|decrement] operation into a dedicated statement.}}
+//                              ^
+            result = val1++ - 1; // Noncompliant
+            result = val2++ * 2; // Noncompliant
+            result = val2++ / 4; // Noncompliant
+            result = --val1 % 2; // Noncompliant
```

Comment:
How about switching the order of the arguments in a couple of the test cases so the literals come first?